includes/delete_datastream.form.inc

```diff
@@ -28,7 +33,18 @@ function islandora_delete_datastream_form(array $form, array &$form_state, Abstr
     t('Cancel')
   );
 }
-
+/**
+ * Purge a given DSID's derivative datastreams on an object.
+ */
+function islandora_datastream_derivatives_purged(AbstractObject $object, $dsid) {
+  module_load_include('inc', 'islandora', 'includes/utilities');
+  $hooks = islandora_invoke_hook_list(ISLANDORA_DERVIATIVE_CREATION_HOOK, $object->models, array($object));
+  $dsid_map = function ($hook_definition) {
+    return $hook_definition['destination_dsid'];
+  };
+  $dsids = array_map($dsid_map, $hooks);
```

Comment:
`destination_dsid` in the definition arrays can be `NULL`, which we should probably strip out (may be handled with the filtering business?) instead of trying to purge a datastream with an ID of `NULL`.